pkg/agent/stats/collector.go

```diff
@@ -204,58 +204,25 @@ func isIdenticalMulticastGroupMap(a, b map[string][]cpv1beta.PodReference) bool
 	return true
 }
 
-// report calculates the delta of the stats and pushes it to the antrea-controller summary API.
-// If multicast feature gate is enabled, it also sends the full multicast group and IGMP report stats to the antrea-controller.
-func (m *Collector) report(curStatsCollection *statsCollection) error {
-	npStats := calculateDiff(curStatsCollection.networkPolicyStats, m.lastStatsCollection.networkPolicyStats)
-	acnpStats := calculateRuleDiff(curStatsCollection.antreaClusterNetworkPolicyStats, m.lastStatsCollection.antreaClusterNetworkPolicyStats)
-	anpStats := calculateRuleDiff(curStatsCollection.antreaNetworkPolicyStats, m.lastStatsCollection.antreaNetworkPolicyStats)
+func (m *Collector) calculateStats(curStatsCollection *statsCollection) (npStats, acnpStats, anpStats []cpv1beta.NetworkPolicyStats) {
+	npStats = calculateDiff(curStatsCollection.networkPolicyStats, m.lastStatsCollection.networkPolicyStats)
+	acnpStats = calculateRuleDiff(curStatsCollection.antreaClusterNetworkPolicyStats, m.lastStatsCollection.antreaClusterNetworkPolicyStats)
+	anpStats = calculateRuleDiff(curStatsCollection.antreaNetworkPolicyStats, m.lastStatsCollection.antreaNetworkPolicyStats)
+	return npStats, acnpStats, anpStats
+}
 
+func (m *Collector) calculateNodeStatsSummary(curStatsCollection *statsCollection) *cpv1beta.NodeStatsSummary {
 	var multicastGroups []cpv1beta.MulticastGroupInfo
-	multicastGroupsUpdated := false
+	npStats, acnpStats, anpStats := m.calculateStats(curStatsCollection)
 	if m.multicastEnabled {
```

Comment:
Still feel better to use a flag `multicastGroupsUpdated` and set its value with function `isIdenticalMulticastGroupMap` inside "if m.multicastEnabled", and then consume the flag in line 222

Reply:
updated